provisioning/miniprov/hare_mp/store.py

```diff
@@ -77,6 +90,31 @@ def __init__(self, url: str, index='hare'):
     def _raw_get(self, key: str) -> str:
         return self.conf.get(self.index, key)
 
+    def _get_nodes_from_services(self, service_type: str, with_hostname=False,
```

Comment:
@rahul27kumar ,  as discussed in the call, let's use just 2 functions, let this function return a list of machine ids corresponding to the service. Rename the same as `get_machine_ids_for_service()` and let `get_nodes_with_hostname()` invoke `get_machine_ids_for_service()` and fetch corresponding hostnames for the received machine ids and return the same. Rename `get_nodes_with_hostname()` to `get_hostnames_for_service()`.